### PR TITLE
Adding Governance docs

### PR DESCRIPTION
**Pull request by diemol**
Governance document proposal for the Selenium project.
The members mentioned already have the quoted role, except the TLC, which are just proposed ones.

**Important**: Code of Conduct is referenced in the document but it has been not created yet. This will come in a different PR.

### CODE REVIEW

**Comment by manoj9788** [comment]
Note: The SFC is reviewing this too! we should hear back from them soon.

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+a Technical Leadership Committee (TLC) which is responsible for a high-level
+technical guidance of the project.
+
+The TLC has final authority over this project including:
+
+* Technical direction
+* Repository hosting
+* Contribution policy (shared with the PLC)
+
+TSC seats are not time-limited. There is no fixed size of the TLC. The TLC
```
Comment (titusfortner) [inline GOVERNANCE.md]: typo

Comment (diemol) [reply GOVERNANCE.md]:
```suggestion
TLC seats are not time-limited. There is no fixed size of the TLC. The TLC
```

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+requests once they have collected the feedback they deem necessary.
+* Discuss once a month the technical status and project roadmap.
+
+To become a TLC member:
+
+* Work in a helpful and collaborative way with the community.
+* Have given good feedback on others' submissions and displayed an overall understanding
+of the code quality standards for the project.
+* Commit to being a part of the community for the long-term.
+* Have submitted a minimum of 20 qualifying pull requests.
```
Comment (diemol) [inline GOVERNANCE.md]:
```suggestion
* Have submitted a minimum of 20 qualifying pull requests or pushed a minimum of 20 qualifying commits across the different projects (IDE, Docker-Selenium, Selenium, Site & Docs).
```

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+There is no expectation of commitment to the project, no specific skill requirements,
+and no selection process.
+
+Some Contributors might have some basic privileges to the GitHub repos, based on
+their type of contribution (e.g., close an issue after triaging it).
+
+Contributors have read-only access to source code and submit changes via pull
+requests. Contributor pull requests have their contribution reviewed and merged
+by a TLC member or a Committer. TLC members and Committers work with Contributors
```
Comment (lukeis) [inline GOVERNANCE.md]: referencing a non-standard acronym before it's defined. Change to -

    [Technical Leadership Committee (TLC)](#technical-leadership-committee-tlc)

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+* Have submitted a minimum of 10 qualifying pull requests. Where significant
+technical weight is present and requires little effort to accept because it is
+well documented and tested.
+
+New Committers can be nominated by any existing Committer or TLC member. Once
+they have been nominated, there will be a vote by the TLC members.
+
+It is important to recognize that having the Committer role is a privilege, not
+a right. That privilege must be earned and once earned, it can be removed by
+the TLC members by a standard TLC motion. However, under normal circumstances
```
Comment (lukeis) [inline GOVERNANCE.md]: from a personal situation perspective (a committer who has not been active in a few years). Should this be a little more explicit on losing the privilege? @shs96c has before said it's a seemingly permanent privilege grant and this seems like it's language for those that might be violating some kind of Code of Conduct?

Like, am I going to just lose it someday unbeknownst to myself until github tells me I've been removed? Or is there plans to have communication with the person? ( "it can be removed by the TLC members by a standard TLC motion" seems not very well defined of a process, especially if one objects?)

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+
+The technical decisions and roadmap of the Selenium project are governed by
+a Technical Leadership Committee (TLC) which is responsible for a high-level
+technical guidance of the project.
+
+The TLC has final authority over this project including:
+
+* Technical direction
+* Repository hosting
+* Contribution policy (shared with the PLC)
```
Comment (lukeis) [inline GOVERNANCE.md]: make PLC a link -

    [PLC](#project-leadership-committee-plc)

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+TLC seats are not time-limited. There is no fixed size of the TLC. The TLC
+should be of such a size as to ensure adequate coverage of important areas
+of expertise balanced with the ability to make decisions efficiently.
+
+The TLC may add additional members to the TLC by a standard TLC motion.
+
+A TLC member may be removed from the TLC by voluntary resignation, or by a stand
+ard TLC motion.
```
Comment (illicitonion) [inline GOVERNANCE.md]: This wrapping inserts a space between "stand" and "ard"

File: GOVERNANCE.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+removal of one or more TLC members affiliated with the over-represented employer(s).
+
+TLC members have additional responsibilities over and above those of a Committer.
+These responsibilities ensure the project has a technical viability and sustainability
+in a smooth way. TLC members are expected to review code contributions, approve
+changes to this document, and manage the copyrights within the project outputs.
+
+TLC members fulfill all requirements of Committers, and also:
+
+* May merge external pull requests for accepted issues upon reviewing and approving
```
Comment (illicitonion) [inline GOVERNANCE.md]: How does this differ from the committers' rights here? Is it because they don't need to escalate to more appropriate owners? If so, it may be worth calling that out explicitly.

Also, what exactly does "external" mean here?

File: GOVERNANCE.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+Leadership Committee (PLC), which acts as a bridge between the Selenium project
+and the Software Freedom Conservancy (SFC).
+
+Since there are many different facets to the Selenium project, the PLC wants to
+reflect more than just people who sling code about. The PLC gets involved in
+different ways, such as whenever the project spends money, enters legal agreements,
+or has to deal with lawyers. Typically the PLC will discuss the topic with the rest
+of the community, and then a voting session will take place.
+
+PLC seats are not time-limited. Only one person may be affiliated with the any
```
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
PLC seats are not time-limited. Only one person affiliated with any
```

File: GOVERNANCE.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+* The existing PLC consults with the TLC and the active Committers, who vote on whether or
+not to add those people.
+    * No-one is ever suggested publicly without first discussing the idea with 
+      them first (to make sure that they are okay with it).
+
+
+The PLC has final authority over this project including:
+
+* Project governance and process (including this policy).
+* Contribution policy (shared with the PLC).
```
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
* Contribution policy (shared with the TLC).
```

File: GOVERNANCE.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+
+A PLC member may be removed from the PLC by voluntary resignation, or by a stand
+ard TSC motion.
+
+#### Process for Adding PLC Members
+
+1. Invite to the Slack PLC chat room (`selenium-plc`)
+1. Tweet congratulations to the new PLC member from the SeleniumHQ Twitter account
+
+The current PLC members are:
```
Comment (illicitonion) [inline GOVERNANCE.md]: Always odd, eh? ;)

File: GOVERNANCE.md [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+The PLC has final authority over this project including:
+
+* Project governance and process (including this policy).
+* Contribution policy (shared with the PLC).
+* Budget and legal related issues.
+
+The PLC also should discuss once a month the overall status of the project and any pending or
+upcoming topics.
+
+A PLC member may be removed from the PLC by voluntary resignation, or by a stand
```
Comment (illicitonion) [inline GOVERNANCE.md]: Bad line wrapping in the middle of a word

File: GOVERNANCE.md [hunk 12]
```diff
@@ -0,0 +1,356 @@
...
+ard TSC motion.
```
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
ard PLC motion.
```

File: GOVERNANCE.md (hunk 6 above)
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
ard TLC or PLC motion.
```
Maybe?

File: GOVERNANCE.md [hunk 13]
```diff
@@ -0,0 +1,356 @@
...
+
+The PLC and TLC follow a [Consensus Seeking](https://en.wikipedia.org/wiki/Consensus-seeking_decision-making)
+decision making model.
+
+When an agenda item has appeared to reach a consensus, the moderator will ask
+"Does anyone object?" as a final call for dissent from the consensus.
+
+If an agenda item cannot reach a consensus, a PLC/TLC member can call for either
+a closing vote or a vote to table the issue to the next meeting. The call for a
+vote must be approved by a majority of the TSC or else the discussion will continue.
```
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
vote must be approved by a majority of the PLC/TSC or else the discussion will continue.
```

File: GOVERNANCE.md [hunk 14]
```diff
@@ -0,0 +1,356 @@
...
+## Commercial and Community Driven Projects
+
+The wide use of the tools provided by the Selenium project has created a large offer of
+commercial services and open source/community driven projects. The Selenium project
+enables and encourages anyone to start any type of project or service that has as its
+objective to simplify and spread the use of Selenium across the community.
+
+Nevertheless, due to its open source, community and non-profit origin, the Selenium
+project will only mention on its website projects of the same nature based on Selenium
+or WebDriver. If a commercial tool or project wants to be mentioned in the Selenium website,
```
Comment (illicitonion) [inline GOVERNANCE.md]:
```suggestion
or WebDriver, except on the sponsors page. If a commercial tool or project wants to be mentioned in the Selenium website,
```

File: GOVERNANCE.md [hunk 15]
```diff
@@ -0,0 +1,356 @@
...
+* One must have shown a willingness and ability to participate in the project as
+a team player.
+* Committers are expected to be respectful of every community member and to work
+collaboratively in the spirit of inclusion.
+* Have submitted a minimum of 10 qualifying pull requests. Where significant
+technical weight is present and requires little effort to accept because it is
+well documented and tested.
+
+New Committers can be nominated by any existing Committer or TLC member. Once
+they have been nominated, there will be a vote by the TLC members.
```
Comment (illicitonion) [inline GOVERNANCE.md]: There seems to be a slight contradiction here in that below it's described that we use consensus-seeking and only use votes if we can't reach consensus, but this says we use a vote. May be worth rephrasing to say that we use the standard consensus-seeking mechanism, rather than a vote?

File: GOVERNANCE.md [hunk 16]
```diff
@@ -0,0 +1,356 @@
...
+given employer can be on the PLC at a time. The PLC member count should be always
+odd so that no votes are ever tied, an ideal minimum number is 5. PLC members
+should also be active members of the community.
+
+To become a PLC member:
+
+* An existing PLC member steps down and suggests someone (or some people) to replace them.
+* Be a committer or a contributor.
+* Commit to being a part of the community for the long-term.
+* The existing PLC consults with the TLC and the active Committers, who vote on whether or
```
Comment (illicitonion) [inline GOVERNANCE.md]: as about wrt consensus-seeking vs voting

File: GOVERNANCE.md (hunk 4 above)
Comment (diemol) [reply GOVERNANCE.md]: As you pointed out, the intent of this section is to remove the committing privilege in case the committer violates some kind of Code of Conduct. Since we do not have it yet, I left it open on purpose (there is a separate stream where the Code of Conduct is being put together).

In general, I don't think that the commit bit should be removed if the committer has been inactive. We cannot just kick out people who have contributed so much from their time to the project, which is in line with what @shs96c has said before. 

Do you think we should make this section more specific? What do you suggest @lukeis?

File: GOVERNANCE.md (hunk 3 above)
Comment (diemol) [reply GOVERNANCE.md]: Good catch, thank you, fixed it.

File: GOVERNANCE.md (hunk 5 above)
Comment (diemol) [reply GOVERNANCE.md]: Good catch, thank you, fixed it.
Feel free to resolve it when you have time to check it.

File: GOVERNANCE.md (hunk 10 above)
Comment (diemol) [reply GOVERNANCE.md]: 😆 
It is because @ahunsberger stepped down and we are looking for one more. Currently, it is being discussed if it is ok to have me and @mmerrell. The dilemma is that we are both employed by Sauce Labs, and... I just realized that adding both of us will keep the number of members even 🤦‍♂ 
We should figure this one out. :)

File: GOVERNANCE.md (hunk 6 above)
Comment (diemol) [reply GOVERNANCE.md]: Good catch, thank you, fixed it.
Feel free to resolve it when you have time to check it.

File: GOVERNANCE.md (hunk 7 above)
Comment (diemol) [reply GOVERNANCE.md]: An external PR is one that comes from (normally) a fork.

Regarding the difference between TLC members and committers rights: yes, the difference is that a TLC should not need to escalate to someone else when merging a PR, but having common sense (e.g. the change is easy enough that anyone can understand it and merge it). Maybe this is too detailed and perhaps we do not need this level of detail, nevertheless, I pushed a change that might be too wordy but it is also more clear.

Please let me know what you think about the change.

File: GOVERNANCE.md (hunk 8 above)
Comment (diemol) [reply GOVERNANCE.md]: I think we wanted to leave the `may be` in case we have two people from the same company...

On the other hand, we could just change this and have the same wording as the `TLC`, which is: `No more than 1/3 of the TLC members may be affiliated with the same employer.`

What do you all think?

File: GOVERNANCE.md (hunk 6 above)
Comment (diemol) [reply GOVERNANCE.md]: I think it is a good idea, I added that.

File: GOVERNANCE.md (hunk 12 above)
Comment (diemol) [reply GOVERNANCE.md]: Fixed

File: GOVERNANCE.md (hunk 14 above)
Comment (diemol) [reply GOVERNANCE.md]: Makes sense.

File: GOVERNANCE.md (hunk 15 above)
Comment (diemol) [reply GOVERNANCE.md]: This is a very good point. In reality, many times we say we are voting but we are actually seeking a consensus. I will change the sections that mention `vote` and use `consensus` instead. We do not want situations where someone is added to a committee because the votation went 5 yes and 4 no.

Comment (diemol) [reply GOVERNANCE.md]: I pushed a change with this, please have a look. Hopefully it looks better now.

File: GOVERNANCE.md (hunk 16 above)
Comment (diemol) [reply GOVERNANCE.md]: I pushed a change with this, please have a look. Hopefully it looks better now.

File: GOVERNANCE.md [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+## Consensus Seeking Process
+
+The PLC and TLC follow a [Consensus Seeking](https://en.wikipedia.org/wiki/Consensus-seeking_decision-making)
+decision making model.
+
+When an agenda item has appeared to reach a consensus, the moderator will ask
+"Does anyone object?" as a final call for dissent from the consensus.
+
+If an agenda item cannot reach a consensus, a PLC/TLC member can call for either
+a closing vote or a vote to table the issue to the next meeting. The call for a
```
Comment (titusfortner) [inline GOVERNANCE.md]: Since the word "table" has different meanings in different countries, maybe we should change it to something like "postpone further discussion on the issue until"

Comment (diemol) [reply GOVERNANCE.md]: Changed it as per the suggestion, please have a look and resolve this conversation if things look good for you @titusfortner.

File: GOVERNANCE.md (hunk 4 above)
Comment (lukeis) [reply GOVERNANCE.md]: I don't have any specific recommendations for now. When a CoC is created / published, this section should be updated with references of that intent.

Thanks for you for confirming the unofficial policy 😄

File: GOVERNANCE.md (hunk 7 above)
Comment (illicitonion) [reply GOVERNANCE.md]: The permissions thing makes sense, thanks :)

I'm still not sure on external - I'd maybe just remove the word?

File: GOVERNANCE.md (hunk 8 above)
Comment (illicitonion) [reply GOVERNANCE.md]: The grammar of the "may be" doesn't quite work with the later "can be" in the sentence...

I'd either phrase as:
"Only one person affiliated with any given employer can be on the PLC at a time."
or
"Only one person on the PLC may be affiliated with any given employer at a time."

File: GOVERNANCE.md (hunk 4 above)
Comment (diemol) [reply GOVERNANCE.md]: 👍

File: GOVERNANCE.md (hunk 7 above)
Comment (diemol) [reply GOVERNANCE.md]: Sure, it will be more understandable. I will remove the word.

File: GOVERNANCE.md (hunk 8 above)
Comment (diemol) [reply GOVERNANCE.md]: Thanks for the suggestion, I added `Only one person on the PLC may be affiliated with any given employer at a time`.

File: GOVERNANCE.md [hunk 18]
```diff
@@ -0,0 +1,364 @@
...
+### Users
+
+Users are community members who have a need for the project. Anyone can be a
+User; there are no special requirements. Common User contributions include:
+
+* Evangelizing the project (e.g., display a link on a website and raise
+awareness through word-of-mouth)
+* Informing strengths and weaknesses from a new user perspective 
+(e.g., raising a bug as a GitHub issue, proposing a new feature)
+* Providing moral support (a "thank you" goes a long way).
```
Comment (ddavison) [inline GOVERNANCE.md]:
```suggestion
* Providing moral support (a "thank you" goes a long way)
```

For consistency with the rest of the bulleted items

File: GOVERNANCE.md [hunk 19]
```diff
@@ -0,0 +1,364 @@
...
+a GitHub issue, and all relevant committers should be pinged so they can weigh
+in on the discussion if desired. 
+* Debates between committers about whether code should be merged should happen
+in GitHub pull requests to keep the project decision history.
+* In general any committer can review and merge a pull request. Committers
+should only merge code they are qualified to review, which might entail pinging
+another committer who has greater ownership over a specific code area.
+* May label and close issues.
+
+To become a Committer:
```
Comment (ddavison) [inline GOVERNANCE.md]: What do you think about making this a sub-heading, for linking purposes?

```suggestion
#### Becoming a committer
```

File: GOVERNANCE.md [hunk 20]
```diff
@@ -0,0 +1,364 @@
...
+
+* One must have shown a willingness and ability to participate in the project as
+a team player.
+* Committers are expected to be respectful of every community member and to work
+collaboratively in the spirit of inclusion.
+* Have submitted a minimum of 10 qualifying pull requests. Where significant
```
Comment (ddavison) [inline GOVERNANCE.md]: This seems a bit arbitrary.  Was this number decided elsewhere?  

What about 

```suggestion
* Have submitted substantive pull requests. Where significant
```

File: GOVERNANCE.md [hunk 21]
```diff
@@ -0,0 +1,364 @@
...
+TLC seats are not time-limited. There is no fixed size of the TLC. The TLC
+should be of such a size as to ensure adequate coverage of important areas
+of expertise balanced with the ability to make decisions efficiently.
+
+The TLC may add additional members to the TLC by a standard TLC motion.
+
+A TLC member may be removed from the TLC by voluntary resignation, or by a 
+standard TLC or PLC motion.
+
+No more than 1/3 of the TLC members may be affiliated with the same employer.
```
Comment (ddavison) [inline GOVERNANCE.md]: 😮 Wow!  👍

Comment (diemol) [reply GOVERNANCE.md]: 👍

File: GOVERNANCE.md (hunk 20 above)
Comment (diemol) [reply GOVERNANCE.md]: The number is a suggestion that I made based on other Governance models that already exist. I strongly think that we need to have a number that can be measured, otherwise things fall into a subjective area. 

Do you think this number makes sense? Should we add `substantive` next to the numer of the number we use?

Comment (ddavison) [reply GOVERNANCE.md]: > otherwise things fall into a subjective area.

I think that's one of the best parts of an open source community.  Your contributions for instance, @diemol -  you didn't have 10 PRs ([you had five](https://github.com/diemol?tab=overview&from=2017-12-01&to=2017-12-31) to [SeleniumHQ](https://github.com/seleniumhq)).  But the PRs you *did* have were substantive and I [recommended you](https://groups.google.com/forum/#!msg/selenium-developers/ovmp6B-OrQA/rXOTubxjAgAJ;context-place=forum/selenium-developers) to be on the team :+1:

It is subjective and IMO it should remain that way.  As long as the contributor is also `+1`'d by other people on the team.

Comment (diemol) [reply GOVERNANCE.md]: You are right, I only had 5 PRs and based on this proposal I should not have been added until I reached the 10 PRs. I also believe that having met the maintainers in person, and already maintaining a similar project helped me to get added.

I honestly think we should have a number defined, but we should reach a consensus. We should also define what `substantive` means. And if we only leave `substantive`, does that mean that one single `substantive` PR is enough qualification? This leaves lots of room for interpretation, and a number takes that away.

Nevertheless, as I said, if the rest of us who are watchers of this PR, are ok with leaving `substantive` instead of a fixed number, please just 👍 this comment. If you prefer to have a fixed number, then please 👎 this comment. If possible please add your arguments.

Comment (titusfortner) [reply GOVERNANCE.md]: This is wrong, working on IDE and Docker containers are part of the official project, just not in the selenium repo. Diego more than qualifies for this position, as does Dave.

I like having an explicit technical requirement, but it's going to be squishy either way since commits/PRs themselves aren't sufficient units of measurement (e.g. if they are all one line comment changes). I think it's fine as is so long as we recognize it as that.

Comment (lukeis) [reply GOVERNANCE.md]: I agree with it not be strictly defined... How about wording like -

Have submitted substantive pull requests (for example a minimum of 10).

Comment (diemol) [reply GOVERNANCE.md]: Ok, how about this? Little wordy but more clear I would say.
(Already submitted the change, but putting it here for simplicity)

```
Have submitted sufficient substantive contributions to one or more of the
different projects (IDE, Docker-Selenium, Selenium, Site & Docs). For
technical contributions, enough weight is present and requires little effort to
accept because it is well documented and tested. Normally 10 substantive
contributions are needed to qualify as a candidate to be a committer, but there
could be cases where the contributions are substantial enough that a fewer
amount is also acceptable.
```

**Comment by shs96c** [comment]
I just had a chat with Bradley from the SFC. We didn't send an email to the `selenium@conservancy` list, so (of course), they haven't had a chance to review it. That's probably my fault. My apologies.

It's the final day of the financial year in the US right now, but hopefully "someone" from the SFC will respond to this PR either here or by email within a week. We can then respond to those comments, and merge. 

Because of this, I'm unilaterally suggesting we push the merge date back a week to 2020-03-10.
